docs/sdk/dev/releases-vs-customization-types.md

```diff
@@ -0,0 +1,366 @@
+        
+        $fooTransfer = $this->fooFacade->findFooByAcme($acmeCriteriaTransfer->getId());
+        $acmeTransfer->setFoo($fooTransfer);
+        
+        return $this->acmExpander->expand($acmeTransfer);   
+    }
+    
+}
+```
+
+And let’s say for some reason, the customer wants to customize `AcmeReader`. For example, some extra logic execution is added.
+
+```php
+class AcmeReader extends SprykerAcmeReader
+{
+    public function readAcme(AcmeCriteriaTransfer $acmeCriteriaTransfer): AcmeTransfer
+    {
+        $acmeTransfer = $this->acmeRepository->findAcmeById($acmeCriteriaTransfer->getId());
+        
+        $fooTransfer = $this->fooFacade->findFooByAcme($acmeCriteriaTransfer->getId());
+        $acmeTransfer->setFoo($fooTransfer);
+        
+        $this->callBeforeExpand($acmeTransfer);
+        
+        $this->acmeExpander->expand($acmeTransfer);
+        
+        return $acmeTransfer;
+    }
+    
+    protected function callBeforeExpand(AcmeTransfer $acmeTransfer): void
+    {
+        // Do something
+    }
+}
+```
+
+Now, let’s imagine Spryker released new versions of `spryker/acme` package.
+
+#### Patch release (e.g., 1.0.0 to 1.0.1).
+
+In version 1.0.1 was fixed a typo in the variables. So this
+
+```php
+return $this->acmExpander->expand($acmeTransfer);
+```
+
+is changed to this
+
+```php
+return $this->acmeExpander->expand($acmeTransfer);
+```
+
+As you can see, it doesn’t affect custom code and everything works fine.
+
+#### Minor release (e.g., 1.0.0 to 1.1.0)
+
+In version 1.1.0 Spryker replaces the method it uses from the `FooFacade`, the method `findFooByAcme(int $id)` is marked as deprecated, instead providing a new method `findFooByCriteria(FooCriteriaTransfer $fooCriteriaTransfer)` with the more generic functionality. The `AcmeReader` also changed and now it looks like:
+
+```php
+public function readAcme(AcmeCriteriaTransfer $acmeCriteriaTransfer): AcmeTransfer
+{
+    $acmeTransfer = $this->acmeRepository->findAcmeById($acmeCriteriaTransfer->getId());
+    
+    $fooCriteriaTransfer = (new FooCriteriaTransfer())
+        ->setAcmeId($acmeCriteriaTransfer->getId();
+    $fooTransfer = $this->fooFacade->findFooByCriteria($fooCriteriaTransfer);
+    $acmeTransfer->setFoo($fooTransfer);
+    
+    return $this->acmExpander->expand($acmeTransfer);   
+}
+```
+
+Does this break customized logic? The answer is no. The project code continues to work just fine using the deprecated external facade method, while it might miss some performance improvements potentially brought by the new method.
+
+#### Major release (e.g., 1.0.0 to 2.0.0)
+
+In major version 2.0.0, the deprecated method `FooFacade::findFooByAcme(int $id)` is completely removed. The upgrader tool will update the `spryker/acme` and `spryker/foo` packages to the latest version, now project code throws `Error` exception and breaks the application because the old method does not exist anymore. This means the developer needs to manually fix this issue.
```

Comment:
```suggestion
In the major version  release 1.0.0 to 2.0.0, the deprecated method `FooFacade::findFooByAcme(int $id)` is completely removed. The upgrader tool will update the `spryker/acme` and `spryker/foo` packages to the latest version. Because of this, the project's code now throws an `Error` exception and breaks the application because the old method does not exist anymore. This means you need to manually fix this issue.
```